packages/core/src/render3/interfaces/definition.ts

```diff
@@ -177,7 +180,7 @@ export interface DirectiveDef<T> extends ɵɵBaseDef<T> {
   /**
    * Factory function used to create a new directive instance.
```

Comment:
Can we add a quick note about why it can be null? e.g.

```suggestion
   * Factory function used to create a new directive instance.
   * Will be null initially. Populated when the factory is first requested by directive instantiation logic.
```